lib/page-object/element.js

```diff
@@ -1,24 +1,169 @@
+ *    @ symbol, false if it does not.
+ */
+Element.prototype.hasElementSelector = function() {
+  return String(this.selector)[0] === '@';
+};
+
+/**
+ * If the element object requires a recursive lookup to resolve its
+ * selector, a new element containing the recursive lookup values
+ * is created and returned.
+ *
+ * @returns {Object} A new Element object containing the element and its
+ *    parents with a recursive lookup strategy for resolving the element
+ *    if one is needed. If not, null is returned.
+ */
+Element.prototype.getRecursiveLookupElement = function () {
+  var lookupList = getAncestorsWithElement(this);
+
+  if (lookupList.length > 1) {
+    return new Element({
+      selector: lookupList,
+      locateStrategy: 'recursion'
+    });
+  }
+
+  return null;
+};
+
+/**
+ * Copies selector properties to the first object from the second if the first
+ * object has undefined values for any of those properties.
+ *
+ * @param {Object} target The object to assign values to.
+ * @param {Object} source The object to capture values from.
+ */
+Element.copyDefaults = function(target, source) {
+  var props = ['name', 'parent', 'selector', 'locateStrategy'];
+  props.forEach(function(prop) {
+    if (target[prop] === undefined || target[prop] === null) {
+      target[prop] = source[prop];
+    }
+  });
+};
+
+/**
+ * Parses the value/selector parameter of an element command creating a
+ * new Element instance with the values it contains, if any. The standard
+ * format for this is a selector string, but additional, complex formats
+ * in the form of an array or object are also supported.
+ *
+ * @param {string|Object} value Selector value to parse into an Element.
+ * @param {string} [using] The using/locateStrategy to use if the selector
+ *    doesn't provide one of its own.
+ */
+Element.fromSelector = function(value, using) {
+
+  if (!value) {
+    throw new Error('Invalid selector value specified');
+  }
+
+  if (value instanceof Element) {
+    value.locateStrategy = value.locateStrategy || using;
+    return value;
+  }
+
+  if (using === 'recursion') {
+    return new Element({
+      selector: value,
+      locateStrategy: 'recursion'
+    });
+  }
+
+  var definition = {};
+
+  if (value && typeof value === 'object') {
+
+    definition.selector = value.selector;
+    if ('locateStrategy' in value) {
+      definition.locateStrategy = value.locateStrategy;
+    }
+
+  } else {
+    definition.selector = value;
+  }
+
+  definition.locateStrategy = definition.locateStrategy || using;
+
+  return new Element(definition);
 };
 
-module.exports = Element;
\ No newline at end of file
+/**
+ * Gets the name of an element instance using "(anonymous)"
+ * if a name was not defined - the case when the element was
+ * created anonymously for a command rather than being a
+ * named element in a page or section definition.
+ */
+function getName(instance) {
```

Comment:
@beatfactor in this PR, you shouldn't unless you imported Element and instantiated some yourself.  In the followup to this one, Element instances are used to parse objects passed into element commands, e.g.

```js
browser.setValue({selector:'input[type=text]', locateStrategy:'css selector'}, 'nightwatch');
```

So when `{selector:'input[type=text]', locateStrategy:'css selector'}` gets parsed internally into an Element, it doesn't have a name to get associated with it so it gets the "anonymous" tag. From there, I don't _think_ there's any messaging that gets exposed to the user that targets the Element instance as a whole, rather, uses the selector and locateStrategy bits from that object as separate values as was the case before. e.g. 

https://github.com/nightwatchjs/nightwatch/pull/1156/files#diff-d6f5178c3f0e3cb65ce5d4fa02c49f4cR350